ambry-tools/src/main/java/com.github.ambry/tools/admin/ServerAdminTool.java

```diff
@@ -0,0 +1,462 @@
+     */
+    @Config("partition.id")
+    @Default("")
+    final String partitionId;
+
+    /**
+     * Path of the file where the data from certain operations will output. For example, the blob from GetBlob and the
+     * user metadata from GetUserMetadata will be written into this file.
+     */
+    @Config("data.output.file.path")
+    @Default("/tmp/ambryResult.out")
+    final String dataOutputFilePath;
+
+    /**
+     * Constructs the configs associated with the tool.
+     * @param verifiableProperties the props to use to load the config.
+     */
+    ServerAdminToolConfig(VerifiableProperties verifiableProperties) {
+      hardwareLayoutFilePath = verifiableProperties.getString("hardware.layout.file.path");
+      partitionLayoutFilePath = verifiableProperties.getString("partition.layout.file.path");
+      typeOfOperation = Operation.valueOf(verifiableProperties.getString("type.of.operation"));
+      hostname = verifiableProperties.getString("hostname", "localhost");
+      port = verifiableProperties.getIntInRange("port", 6667, 1, 65535);
+      blobId = verifiableProperties.getString("blob.id", "");
+      getOption = GetOption.valueOf(verifiableProperties.getString("get.option", "None"));
+      partitionId = verifiableProperties.getString("partition.id", "");
+      dataOutputFilePath = verifiableProperties.getString("data.output.file.path", "/tmp/ambryResult.out");
+    }
+  }
+
+  /**
+   * Runs the server admin tool
+   * @param args associated arguments.
+   * @throws Exception
+   */
+  public static void main(String args[]) throws Exception {
+    VerifiableProperties verifiableProperties = ToolUtils.getVerifiableProperties(args);
+    ServerAdminToolConfig config = new ServerAdminToolConfig(verifiableProperties);
+    ClusterMapConfig clusterMapConfig = new ClusterMapConfig(verifiableProperties);
+    ClusterMap clusterMap =
+        ((ClusterAgentsFactory) Utils.getObj(clusterMapConfig.clusterMapClusterAgentsFactory, clusterMapConfig,
+            config.hardwareLayoutFilePath, config.partitionLayoutFilePath)).getClusterMap();
+    SSLFactory sslFactory = !clusterMapConfig.clusterMapSslEnabledDatacenters.isEmpty() ? new SSLFactory(
+        new SSLConfig(verifiableProperties)) : null;
+    ServerAdminTool serverAdminTool =
+        new ServerAdminTool(clusterMap.getMetricRegistry(), sslFactory, verifiableProperties);
+    File file = new File(config.dataOutputFilePath);
+    if (!file.exists() && !file.createNewFile()) {
+      throw new IllegalStateException("Could not create " + file);
+    }
+    FileOutputStream outputFileStream = new FileOutputStream(config.dataOutputFilePath);
+    DataNodeId dataNodeId = clusterMap.getDataNodeId(config.hostname, config.port);
+    if (dataNodeId == null) {
+      throw new IllegalArgumentException(
+          "Could not find a data node corresponding to " + config.hostname + ":" + config.port);
+    }
+    switch (config.typeOfOperation) {
```

Comment:
Tried all four operations, ensured that blob data and user metadata written to output file matches what was put and tried `GetOptions.None` and `GetOptions.Include_All`.